internal/schema/0.12/variable_block.go

```diff
@@ -0,0 +1,63 @@
+package schema
+
+import (
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func variableBlockSchema(v *version.Version) *schema.BlockSchema {
+	bs := &schema.BlockSchema{
+		Labels: []*schema.LabelSchema{
+			{
+				Name:        "name",
+				Description: lang.PlainText("Variable Name"),
+			},
+		},
+		Description: lang.Markdown("Input variable allowing users to customizate aspects of the configuration when used directly " +
+			"(e.g. via CLI, `tfvars` file or via environment variables), or as a module (via `module` arguments)"),
+		Body: &schema.BodySchema{
+			Attributes: map[string]*schema.AttributeSchema{
+				"description": {
+					ValueType:   cty.String,
```

Comment:
the Terraform SDK was hesitant to expose the `cty` namespace (or rather, commit to supporting cty as the typesystem for the SDK) so we forked and owned a hashicorp namespaced "cty". I'm assuming though, this project shouldn't concern itself the same way?

Because it's a very specific library, and there could be some reintegration with TF core down the road (meaning we should work in Cty).

Just bringing it up, nothing of concern here, other than exposing cty to the API surface

Reply:
🤔  That's certainly a good question to ask.

The library is entirely decoupled from Terraform for now and I don't know if it will integrate with core in any way at this point, but it's possible that in the future we may want to move [functions](https://github.com/hashicorp/terraform/blob/master/lang/functions.go) here and these make very heavy use of cty, so 100% compatibility may be beneficial, if not necessary.

As far as I remember the SDK's main motivation behind using the fork was that SDK has a high number of consumers (providers) and so any change in dependencies and generally anything that can be perceived as breaking change can be painful to overcome.

I don't expect this particular library (`terraform-schema`) to have _that_ many consumers beyond language server, certainly not as many as the SDK.

This _would_ be a valid question to ask about `hcl-lang`, but even there I would probably lean towards keeping the dependency as is, just because I think about `hcl-lang` as a complimentary library to `hcl` itself, which [does depend on `zclconf/go-cty`](https://github.com/hashicorp/hcl/blob/3de61ecba2981c715584071070bd18723cf7299d/go.mod#L19).